### PR TITLE
108 install and configure jest unit testing

### LINKED ISSUE #108: Install and configure Jest for webpacker unit testing
### Description

In order to have unit test coverage for webpacker javascript, we're switching from karma in favor of jest (https://facebook.github.io/jest/) Jest seems better suited for ES6 js, has more built-in features (coverage reporting, mocking, etc) and seems much more easily configureable.

### Done Looks Like

- Jest is installed, configured.
- Basic, initial unit testing of ES6 classes and function mocks are in place.

### Tasks
- [x] Install Jest and necessary configs for babel and importing.
- [x] Research best pattern for isolated unit testing.
- [x] Set up initial tests.

### PR DESCRIPTION
**Pull request by adamjarling**
Fixes #108 

For webpacker unit testing, removes Karma and replaces with Jest.   Wired up a few sample tests (which will get moved around), which show how Jest unit tests can test in isolation and how to mock API calls and `node_modules`.

To run the tests locally:
```
yarn test
```

@samvera-labs/avalon

### CODE REVIEW

File: app/javascript/avalon-player/AvalonPlayer.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import * as avalonApi from '../avalon-api';
+
+// Mock network requests by Jest convention
+jest.mock('../avalon-api');
+
+test('Fetches a manifest only if a valid media object id is supplied', async () => {
+  const playerInstance = new AvalonPlayer();
+  let manifest = await playerInstance.getManifest('123abc');
+  expect(manifest.id).toEqual(
+    'http://localhost:3333/concern/generic_works/3484zg88m/manifest'
```
Comment (adamjarling) [inline app/javascript/avalon-player/AvalonPlayer.test.js]: I need to update this to the `window` variable...

File: config/webpack/test.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+// environment.loaders.set('istanbul-instrumenter', {
+//   test: /\.js$/,
+//   enforce: "post",
+//   loader: "istanbul-instrumenter-loader",
+//   query: {
+//     esModules: true
+//   },
+//   exclude: ["node_modules", /\.spec\.js$/]
+// }) // optional
+
```
Comment (adamjarling) [inline config/webpack/test.js]: Ah, @cjcolvar was this the stuff that was causing the Circle CI build to break?   Previous karma config from webpacker docs that I put in and must have missed when cleaning up... if so.

Comment (cjcolvar) [reply config/webpack/test.js]: Yeah, my commit fixed the Circle CI build.  I used this issue as a hint: https://github.com/webpack/webpack/issues/4569.

File: app/javascript/avalon-player/AvalonPlayer.test.js (hunk 1 above)
Comment (adamjarling) [reply app/javascript/avalon-player/AvalonPlayer.test.js]: Nevermind, it's a mock object, doesn't matter what this value is...

**Comment by adamjarling** [comment]
I personally have a very strong preference for keeping a `something.test.js` test file living directly next to it's file `something.js`.  Otherwise in my experience, they get painfully neglected, or never created.  That work for you?

**Comment by adamjarling** [comment]
And I think by organizing and having the tests live in `/javascripts`, it'll keep our javascripts away from Hyrax javascripts.  

It's Jest convention to generally follow this pattern, or put all tests in a `__tests__` directory.

**Pull request merged**